src/ui/setup/helpers.ts

```diff
@@ -28,6 +32,22 @@ export async function setupAppHelper(store: UIStore) {
   const recordingId = getRecordingId();
   const replaySession = recordingId ? await getReplaySession(recordingId) : undefined;
 
+  window.addEventListener("unhandledrejection", function (event) {
+    const { reason } = event;
+
+    if (
+      reason &&
+      (isCommandError(reason, ProtocolError.UnknownSession) ||
+        isCommandError(reason, ProtocolError.SessionDestroyed))
```

Comment:
We could handle these errors in `socket.ts`, that would also not only handle them when they're uncaught.

Reply:
Can you be more concrete about what you're proposing?

I thought about handling them in `socket.ts` first, but then I would have to add another module-level event listener for that specific error type, so I just did this instead. I'm not strongly opposed to moving them back into there, but I don't love the `gEventListeners`/`gEventToCallbacksMap` approach and didn't want to lean more into it.

Reply:
Tell me if you like https://github.com/replayio/devtools/pull/10353/commits/9bd1aed7d72d53ba29b7d7f989cb57d17ea698ea better.

Reply:
That additonal listener is unfortunate but I still prefer this to be handled in `socket.ts`.

Reply:
Cool. I have no strong preference here so let's go with that then.